source/CCustomOpcodeSystem.cpp

```diff
@@ -1821,19 +1807,30 @@ extern "C"
 		*thread << value;
 	}
 
-	LPSTR WINAPI CLEO_ReadStringOpcodeParam(CLEO::CRunningScript* thread, char *buf, int size)
+	LPCSTR WINAPI CLEO_ReadStringOpcodeParam(CLEO::CRunningScript* thread, char* buff, int buffSize)
 	{
-		return CLEO_ReadStringPointerOpcodeParam(thread, buf, size); // always support all string param types
+		static char internal_buff[MAX_STR_LEN + 1]; // and terminator
+		if (!buff) 
+		{
+			buff = internal_buff;
+			buffSize = (buffSize > 0) ? min(buffSize, sizeof(internal_buff)) : sizeof(internal_buff); // allow user's length limit
+		}
+
+		auto result = CLEO_ReadStringPointerOpcodeParam(thread, buff, buffSize);

```

Comment:
call ReadStringParam directly to skip unnecessary buf check?

Reply:
fixed